Stuck Quarkus process when port 8080 is already assigned, kill -9 needs to be used, user not informed about used port

Stuck Quarkus process when port 8080 is already assigned, `kill -9` needs to be used, user not informed about used port.

This is regression comparing to 0.22.0 when user is informed via `Caused by: java.net.BindException: Address already in use` and user can press Ctrl+C to stop the process.

Master based app doesn't react to Ctrl+C and `kill -9` needs to be used.
Stacktrace informs just about NPE, nothing about already used port.

```
java -jar getting-started-two/target/getting-started-two-1.0-SNAPSHOT-runner.jar
2019-09-20 09:37:26,984 SEVERE [io.ver.cor.imp.ContextImpl] (vert.x-eventloop-thread-15) Unhandled exception: java.lang.NullPointerException
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$WebDeploymentVerticle.lambda$start$0(VertxHttpRecorder.java:382)
	at io.vertx.core.http.impl.HttpServerImpl.lambda$null$5(HttpServerImpl.java:277)
	at io.vertx.core.impl.ContextImpl.executeTask(ContextImpl.java:369)
	at io.vertx.core.impl.EventLoopContext.lambda$executeAsync$0(EventLoopContext.java:38)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:416)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:515)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:918)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)

2019-09-20 09:37:26,984 SEVERE [io.ver.cor.imp.ContextImpl] (vert.x-eventloop-thread-9) Unhandled exception: java.lang.NullPointerException
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$WebDeploymentVerticle.lambda$start$0(VertxHttpRecorder.java:382)
	at io.vertx.core.http.impl.HttpServerImpl.lambda$null$5(HttpServerImpl.java:277)
	at io.vertx.core.impl.ContextImpl.executeTask(ContextImpl.java:369)
	at io.vertx.core.impl.EventLoopContext.lambda$executeAsync$0(EventLoopContext.java:38)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:416)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:515)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:918)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)

2019-09-20 09:37:26,984 SEVERE [io.ver.cor.imp.ContextImpl] (vert.x-eventloop-thread-1) Unhandled exception: java.lang.NullPointerException
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$WebDeploymentVerticle.lambda$start$0(VertxHttpRecorder.java:382)
	at io.vertx.core.http.impl.HttpServerImpl.lambda$null$5(HttpServerImpl.java:277)
	at io.vertx.core.impl.ContextImpl.executeTask(ContextImpl.java:369)
	at io.vertx.core.impl.EventLoopContext.lambda$executeAsync$0(EventLoopContext.java:38)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:416)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:515)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:918)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)

2019-09-20 09:37:26,984 SEVERE [io.ver.cor.imp.ContextImpl] (vert.x-eventloop-thread-8) Unhandled exception: java.lang.NullPointerException
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$WebDeploymentVerticle.lambda$start$0(VertxHttpRecorder.java:382)
	at io.vertx.core.http.impl.HttpServerImpl.lambda$null$5(HttpServerImpl.java:277)
	at io.vertx.core.impl.ContextImpl.executeTask(ContextImpl.java:369)
	at io.vertx.core.impl.EventLoopContext.lambda$executeAsync$0(EventLoopContext.java:38)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:416)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:515)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:918)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)

2019-09-20 09:37:26,984 SEVERE [io.ver.cor.imp.ContextImpl] (vert.x-eventloop-thread-4) Unhandled exception: java.lang.NullPointerException
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$WebDeploymentVerticle.lambda$start$0(VertxHttpRecorder.java:382)
	at io.vertx.core.http.impl.HttpServerImpl.lambda$null$5(HttpServerImpl.java:277)
	at io.vertx.core.impl.ContextImpl.executeTask(ContextImpl.java:369)
	at io.vertx.core.impl.EventLoopContext.lambda$executeAsync$0(EventLoopContext.java:38)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:416)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:515)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:918)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)

2019-09-20 09:37:26,984 SEVERE [io.ver.cor.imp.ContextImpl] (vert.x-eventloop-thread-14) Unhandled exception: java.lang.NullPointerException
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$WebDeploymentVerticle.lambda$start$0(VertxHttpRecorder.java:382)
	at io.vertx.core.http.impl.HttpServerImpl.lambda$null$5(HttpServerImpl.java:277)
	at io.vertx.core.impl.ContextImpl.executeTask(ContextImpl.java:369)
	at io.vertx.core.impl.EventLoopContext.lambda$executeAsync$0(EventLoopContext.java:38)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:416)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:515)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:918)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)

2019-09-20 09:37:26,984 SEVERE [io.ver.cor.imp.ContextImpl] (vert.x-eventloop-thread-12) Unhandled exception: java.lang.NullPointerException
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$WebDeploymentVerticle.lambda$start$0(VertxHttpRecorder.java:382)
	at io.vertx.core.http.impl.HttpServerImpl.lambda$null$5(HttpServerImpl.java:277)
	at io.vertx.core.impl.ContextImpl.executeTask(ContextImpl.java:369)
	at io.vertx.core.impl.EventLoopContext.lambda$executeAsync$0(EventLoopContext.java:38)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:416)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:515)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:918)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)

2019-09-20 09:37:26,984 SEVERE [io.ver.cor.imp.ContextImpl] (vert.x-eventloop-thread-10) Unhandled exception: java.lang.NullPointerException
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$WebDeploymentVerticle.lambda$start$0(VertxHttpRecorder.java:382)
	at io.vertx.core.http.impl.HttpServerImpl.lambda$null$5(HttpServerImpl.java:277)
	at io.vertx.core.impl.ContextImpl.executeTask(ContextImpl.java:369)
	at io.vertx.core.impl.EventLoopContext.lambda$executeAsync$0(EventLoopContext.java:38)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:416)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:515)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:918)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)

2019-09-20 09:37:26,984 SEVERE [io.ver.cor.imp.ContextImpl] (vert.x-eventloop-thread-11) Unhandled exception: java.lang.NullPointerException
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$WebDeploymentVerticle.lambda$start$0(VertxHttpRecorder.java:382)
	at io.vertx.core.http.impl.HttpServerImpl.lambda$null$5(HttpServerImpl.java:277)
	at io.vertx.core.impl.ContextImpl.executeTask(ContextImpl.java:369)
	at io.vertx.core.impl.EventLoopContext.lambda$executeAsync$0(EventLoopContext.java:38)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:416)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:515)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:918)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)

2019-09-20 09:37:26,984 SEVERE [io.ver.cor.imp.ContextImpl] (vert.x-eventloop-thread-3) Unhandled exception: java.lang.NullPointerException
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$WebDeploymentVerticle.lambda$start$0(VertxHttpRecorder.java:382)
	at io.vertx.core.http.impl.HttpServerImpl.lambda$null$5(HttpServerImpl.java:277)
	at io.vertx.core.impl.ContextImpl.executeTask(ContextImpl.java:369)
	at io.vertx.core.impl.EventLoopContext.lambda$executeAsync$0(EventLoopContext.java:38)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:416)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:515)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:918)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)

2019-09-20 09:37:26,984 SEVERE [io.ver.cor.imp.ContextImpl] (vert.x-eventloop-thread-13) Unhandled exception: java.lang.NullPointerException
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$WebDeploymentVerticle.lambda$start$0(VertxHttpRecorder.java:382)
	at io.vertx.core.http.impl.HttpServerImpl.lambda$null$5(HttpServerImpl.java:277)
	at io.vertx.core.impl.ContextImpl.executeTask(ContextImpl.java:369)
	at io.vertx.core.impl.EventLoopContext.lambda$executeAsync$0(EventLoopContext.java:38)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:416)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:515)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:918)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)

2019-09-20 09:37:26,984 SEVERE [io.ver.cor.imp.ContextImpl] (vert.x-eventloop-thread-6) Unhandled exception: java.lang.NullPointerException
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$WebDeploymentVerticle.lambda$start$0(VertxHttpRecorder.java:382)
	at io.vertx.core.http.impl.HttpServerImpl.lambda$null$5(HttpServerImpl.java:277)
	at io.vertx.core.impl.ContextImpl.executeTask(ContextImpl.java:369)
	at io.vertx.core.impl.EventLoopContext.lambda$executeAsync$0(EventLoopContext.java:38)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:416)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:515)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:918)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)

2019-09-20 09:37:26,984 SEVERE [io.ver.cor.imp.ContextImpl] (vert.x-eventloop-thread-5) Unhandled exception: java.lang.NullPointerException
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$WebDeploymentVerticle.lambda$start$0(VertxHttpRecorder.java:382)
	at io.vertx.core.http.impl.HttpServerImpl.lambda$null$5(HttpServerImpl.java:277)
	at io.vertx.core.impl.ContextImpl.executeTask(ContextImpl.java:369)
	at io.vertx.core.impl.EventLoopContext.lambda$executeAsync$0(EventLoopContext.java:38)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:416)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:515)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:918)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)

2019-09-20 09:37:26,984 SEVERE [io.ver.cor.imp.ContextImpl] (vert.x-eventloop-thread-0) Unhandled exception: java.lang.NullPointerException
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$WebDeploymentVerticle.lambda$start$0(VertxHttpRecorder.java:382)
	at io.vertx.core.http.impl.HttpServerImpl.lambda$null$5(HttpServerImpl.java:277)
	at io.vertx.core.impl.ContextImpl.executeTask(ContextImpl.java:369)
	at io.vertx.core.impl.EventLoopContext.lambda$executeAsync$0(EventLoopContext.java:38)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:416)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:515)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:918)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)

2019-09-20 09:37:26,984 SEVERE [io.ver.cor.imp.ContextImpl] (vert.x-eventloop-thread-2) Unhandled exception: java.lang.NullPointerException
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$WebDeploymentVerticle.lambda$start$0(VertxHttpRecorder.java:382)
	at io.vertx.core.http.impl.HttpServerImpl.lambda$null$5(HttpServerImpl.java:277)
	at io.vertx.core.impl.ContextImpl.executeTask(ContextImpl.java:369)
	at io.vertx.core.impl.EventLoopContext.lambda$executeAsync$0(EventLoopContext.java:38)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:416)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:515)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:918)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)

2019-09-20 09:37:26,984 SEVERE [io.ver.cor.imp.ContextImpl] (vert.x-eventloop-thread-7) Unhandled exception: java.lang.NullPointerException
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$WebDeploymentVerticle.lambda$start$0(VertxHttpRecorder.java:382)
	at io.vertx.core.http.impl.HttpServerImpl.lambda$null$5(HttpServerImpl.java:277)
	at io.vertx.core.impl.ContextImpl.executeTask(ContextImpl.java:369)
	at io.vertx.core.impl.EventLoopContext.lambda$executeAsync$0(EventLoopContext.java:38)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:416)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:515)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:918)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)

Killed: 9
```

Steps to reproduce:
```

mvn io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:create \
    -DprojectGroupId=org.acme \
    -DprojectArtifactId=getting-started-one \
    -DclassName="org.acme.quickstart.GreetingResource" \
    -Dpath="/hello"


mvn io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:create \
    -DprojectGroupId=org.acme \
    -DprojectArtifactId=getting-started-two \
    -DclassName="org.acme.quickstart.GreetingResource" \
    -Dpath="/hello"


mvn clean package -f getting-started-one/pom.xml
mvn clean package -f getting-started-two/pom.xml

java -jar getting-started-one/target/getting-started-one-1.0-SNAPSHOT-runner.jar &
java -jar getting-started-two/target/getting-started-two-1.0-SNAPSHOT-runner.jar
```

Run following command in new terminal to kill the stuck process
```
jps -l | grep started-two | cut -d" " -f1 | xargs kill -9
```

Don't forget to stop the first app
```
jps -l | grep started-one | cut -d" " -f1 | xargs kill
```

--- a/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java
+++ b/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java
@@ -385,19 +385,27 @@ public class VertxHttpRecorder {
             httpServer = vertx.createHttpServer(httpOptions);
             httpServer.requestHandler(router);
             httpServer.listen(port, host, event -> {
-                // Port may be random, so set the actual port
-                httpOptions.setPort(event.result().actualPort());
-                if (remainingCount.decrementAndGet() == 0) {
-                    startFuture.complete(null);
+                if (event.cause() != null) {
+                    startFuture.fail(event.cause());
+                } else {
+                    // Port may be random, so set the actual port
+                    httpOptions.setPort(event.result().actualPort());
+                    if (remainingCount.decrementAndGet() == 0) {
+                        startFuture.complete(null);
+                    }
                 }
             });
             if (httpsOptions != null) {
                 httpsServer = vertx.createHttpServer(httpsOptions);
                 httpsServer.requestHandler(router);
                 httpsServer.listen(httpsPort, host, event -> {
-                    httpsOptions.setPort(event.result().actualPort());
-                    if (remainingCount.decrementAndGet() == 0) {
-                        startFuture.complete();
+                    if (event.cause() != null) {
+                        startFuture.fail(event.cause());
+                    } else {
+                        httpsOptions.setPort(event.result().actualPort());
+                        if (remainingCount.decrementAndGet() == 0) {
+                            startFuture.complete();
+                        }
                     }
                 });
             }
@@ -476,4 +484,5 @@ public class VertxHttpRecorder {
     public static Router getRouter() {
         return router;
     }
+
 }

